src/router/Route.ts

```diff
@@ -29,6 +33,19 @@ export default class Route extends Component<IRouteProps, any> {
 				onEnter({ props: this.props, router });
 			});
 		}
+
+		const { getComponent } = this.props;
+		if (getComponent) {
+			Promise.resolve().then(() => {
+				getComponent({ props: this.props, router }, this.onComponentResolved);
+			});
+		}
+	}
+
+	onComponentResolved = (error, component) => {
```

Comment:
The error is to mirror react-router [here](https://github.com/ReactTraining/react-router/blob/master/docs/API.md#callback-signature)

I don't know how heavily you want to mirror them, but I added it in just in case.